src/pages/AddBook.jsx

```diff
@@ -0,0 +1,130 @@
+        const requestOptions = {
+            method: "POST",
+            mode: 'cors',
+            headers: {"Content-Type": "application/json",
+                "Accept": "application/json"
+            },
+            body:JSON.stringify(data)
+        }
+
+        fetch('https://book-swap-api.dev.io-academy.uk/api/books', requestOptions)
+            .then(response => response.json())
+            .then(data => {
+                if (data.errors) {    
+                    setTitleError(data.errors.title)                           
+                    setAuthorError(data.errors.author)                                 
+                    setGenreError(data.errors.genre_id)
+                    setImageURLError(data.errors.image)
+                    setPageCountError(data.errors.page_count)
+                    setYearError(data.errors.year)
+                } else {
+                    setSuccessMessage(data.message)
+                }           
+                    
+            })
+    }
+
+    function handleSubmit(e) {
+        e.preventDefault()
+        addBook()       
+    }
+
+    function handleTitleChange(e) {
+        setTitle(e.target.value)
+    }
+
+    function handleAuthorChange(e) {
+        setAuthor(e.target.value)
+    }
+
+    function handleGenreChange(e) {
+        setGenre(e.target.value)
+    }
+
+    function handleYearChange(e) {
+        setYear(e.target.value)
+    }
+
+    function handlePageCountChange(e) {
+        setPageCount(e.target.value)
+    }
+
+    function handleImageURLChange(e) {
+        setImageURL(e.target.value)
+    }
+
+    function handleBlurbChange(e) {
+        setBlurb(e.target.value)
+    }
+
+    return (
+        <div className="flex justify-center">
+            <form onSubmit={handleSubmit} 
+            className="flex flex-col p-4 w-xl">            
+                <label className="mt-2 mb-1" htmlFor="title">Title (required)</label>
+                <Highlighted text={titleError} />
+                <input onChange={handleTitleChange} className="px-1 border-[1px]" type="text" placeholder="Title" name="title" id="title" />
+                <label className="mt-2 mb-1" htmlFor="author">Author (required)</label>
+                <Highlighted text={authorError} />
+                <input onChange={handleAuthorChange} className="px-1 border-[1px]" type="text" placeholder="Author" name="author" id="author" />
+                <label className="mt-2 mb-1" htmlFor="genre">Genre (required)</label>
+                <Highlighted text={genreError} />
+                <select id="genre" aria-label="Select genre" className="px-1 border-[1px]" onChange={handleGenreChange}>
+                    <option value="">None Selected</option>
+                    <option value="1">Thriller</option>
+                    <option value="2">Romance</option>
+                    <option value="3">Historical</option>
+                    <option value="4">Non-fiction</option>
+                </select>
```

Comment:
There's a better solution here - new genres could be added to the API at any time, so you'd have to be constantly updating these options to try and keep them in sync.

However, the API is able to give you a list of genres that you could use to generate these options instead.